### ISSUE TITLE
Nil pointer dereference in policy.(r *rule).matches method

### ISSUE THREAD

**Issue by brb** [cilium/cilium#7968]
A user reported the following panic with Cilium v1.5.0:

```
level=info msg="Policy imported via API, recalculating..." policyAddRequest=e47b14f2-722e-11e9-a99f-fa163e3c9453 policyRevision=159 subsys=daemon
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2302c50]

goroutine 979 [running]:
github.com/cilium/cilium/pkg/policy.(*rule).matches(0xc0005b09a0, 0x173, 0x0, 0xc001d12100)
        /go/src/github.com/cilium/cilium/pkg/policy/rule.go:469 +0xb0
github.com/cilium/cilium/pkg/policy.ruleSlice.updateEndpointsCaches(0xc000a3e380, 0x70, 0x70, 0x33c9ba0, 0xc001dfab40, 0xc00212aae0, 0x7f262c1e1a58)
        /go/src/github.com/cilium/cilium/pkg/policy/rules.go:284 +0xac
github.com/cilium/cilium/pkg/policy.ruleSlice.UpdateRulesEndpointsCaches.func1(0x33c9ba0, 0xc001dfab40)
        /go/src/github.com/cilium/cilium/pkg/policy/repository.go:462 +0x66
github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEach.func1(0xc002270bd0, 0xc00181dbd4, 0x33c9ba0, 0xc001dfab40)
        /go/src/github.com/cilium/cilium/pkg/policy/identifier.go:75 +0x3a
created by github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEach
        /go/src/github.com/cilium/cilium/pkg/policy/identifier.go:74 +0x12c
level=fatal msg="Agent pipe unexpectedly closed, shutting down" subsys=cilium-node-monitor
```

From inspecting the stack trace, we can see that `securityIdentity` was nil as the time it was accessed in (`offset(securityIdentity.LabelArray)` = 0x20 == the addr from the stack trace, so clearly `securityIdentity` is nil):

https://github.com/cilium/cilium/blob/fdf36ec8c203030d97cbd8b47e2762519ce95997/pkg/policy/rule.go#L469

@tgraf suggested to either assign an init identity (we already do this: https://github.com/cilium/cilium/commit/85e25bcc22bdcc0163a7b581de14cd7658caf69d) or to audit all code.

**Comment by aanm** [comment]
@brb this is already fixed in v1.5 https://github.com/cilium/cilium/pull/7890

**Comment by brb** [comment]
@aanm Could you refer to the commit which fixes this?

**Comment by aanm** [comment]
@brb https://github.com/cilium/cilium/pull/7890/commits/b199f9652dfa4d7d7d6132304f909e3309701244#diff-be710a140ead42aef83a0fb1166eb701R288 (note that's not the upstream commit)

**Comment by brb** [comment]
Fixed in v1.5.1.
